be/src/storage/tablet_updates.cpp

```diff
@@ -2971,6 +2981,18 @@ void TabletUpdates::_remove_unused_rowsets() {
     }
 }
 
+Status TabletUpdates::check_and_remove_rowset() {
+    std::lock_guard l1(_lock);
```

Comment:
why adding lock?

Reply:
> why adding lock?

The lock is to protect the remove operation. But it seems the lock is not necessary, it has been removed.